loaders/httpfs/loader.go

```diff
@@ -0,0 +1,30 @@
+package httpfs
+
+import (
+	"io"
+	"net/http"
+)
+
+type httpFileSystemLoader struct {
+	http.FileSystem
+}
+
+func NewLoader(fs http.FileSystem) httpFileSystemLoader {
```

Comment:
Some resources that I googled right now:

```
A great rule of thumb for Go is accept interfaces, return structs.
–Jack Lindamood
```

https://dave.cheney.net/2016/08/20/solid-go-design
http://idiomaticgo.com/post/best-practice/accept-interfaces-return-structs/
https://medium.com/@cep21/what-accept-interfaces-return-structs-means-in-go-2fe879e25ee8#.ervtbgrdp
https://medium.com/@cep21/preemptive-interface-anti-pattern-in-go-54c18ac0668a#.d0zdjc6pm

Pls let me know if I should change the return type or not. I can of course do it anyway, if you want.